src/get_context_sequences.py

```diff
@@ -0,0 +1,215 @@
+    return text_seq_string
+
+####    functions to create context sequences from multiple strings ####
+def concat_sequences_left(previous_sequence, current_sequence, target_length = 120):
+    # if previous sequence is long, we want it to be no more than half of context sequence
+    max_previous_length = target_length//2
+    
+    # cleaning up sequences
+    previous_sequence = cleanup_segments(str(previous_sequence))
+    current_sequence = cleanup_segments(str(current_sequence))
+    
+    
+    previous_as_list = re.split(r'([.!?])', previous_sequence)
+    if (previous_as_list[-1] == '') & (len(previous_as_list) != 1):
+        prev_last_sentence = previous_as_list[-3:]
+        prev_last_sentence = ''.join(prev_last_sentence)
+    else:
+        prev_last_sentence = previous_as_list[-1]
+
+    # only keep last part of previous sequence if its longer than half of the max sequence length in bert model
+    prev_last_sentence_as_list = prev_last_sentence.split(' ')
+    n_words = len(prev_last_sentence_as_list)
+    if n_words > max_previous_length:
+        prev_last_sentence_as_list = prev_last_sentence_as_list[-max_previous_length:]
+        prev_last_sentence = ' '.join(prev_last_sentence_as_list)
+    # use new line (/n) as token to signify where current sequence begings
+    return prev_last_sentence + ' /n ' + current_sequence
+
+def concat_sequences_full(previous_sequence, current_sequence, next_sequence, target_length = 120):
+    # if previous sequence is long, we want it to be no more than half of context sequence
+    max_previous_length = target_length//3
+    
+    # cleaning up sequences
+    previous_sequence = cleanup_segments(str(previous_sequence))
+    current_sequence = cleanup_segments(str(current_sequence))
+    next_sequence = cleanup_segments(str(next_sequence))
+    
+    
+    previous_as_list = re.split(r'([.!?])', previous_sequence)
+    if (previous_as_list[-1] == '') & (len(previous_as_list) != 1):
+        prev_last_sentence = previous_as_list[-3:]
+        prev_last_sentence = ''.join(prev_last_sentence)
+    else:
+        prev_last_sentence = previous_as_list[-1]
+        
+    next_as_list = re.split(r'([.!?])', next_sequence)
+    if len(next_as_list) != 1:
+        next_first_sentence = next_as_list[:2]
+        next_first_sentence = ''.join(next_first_sentence)
+    else:
+        next_first_sentence = next_as_list[0]
+
+    # only keep last part of previous sequence if its longer than a third of the max sequence length in bert model
+    prev_last_sentence_as_list = prev_last_sentence.split(' ')
+    n_words = len(prev_last_sentence_as_list)
+    if n_words > max_previous_length:
+        prev_last_sentence_as_list = prev_last_sentence_as_list[-max_previous_length:]
+        prev_last_sentence = ' '.join(prev_last_sentence_as_list)
+    # use new line (/n) as token to signify where current sequence begings
+    return prev_last_sentence + ' /n ' + current_sequence + ' /n ' + next_first_sentence
```

Comment:
"\n" is the standard symbol. However, you have a special token for this, right? Then it doesn't really matter and we keep it this way.

Reply:
I don't think it's necessary right now. There are no other "/"'s in the data anyway, so it does not make much of a difference.

But we can try it later.